File: core/src/com/inkostilation/pong/commands/request/RequestConnectionCommand.java
package com.inkostilation.pong.commands.request;

import com.inkostilation.pong.commands.AbstractPongCommand;
import com.inkostilation.pong.engine.IPongEngine;

import java.io.IOException;

public class RequestConnectionCommand<M> extends AbstractPongCommand<M> {

    private int index;

    public RequestConnectionCommand(int index) {
        this.index = index;
    }

    @Override
    protected void execute(IPongEngine<M> engine) throws IOException {
        engine.connectToGame(getMarker(), index);
    }
}

File: core/src/com/inkostilation/pong/commands/request/RequestConnectionCommand.kt
package com.inkostilation.pong.commands.request

import com.inkostilation.pong.commands.AbstractPongCommand
import com.inkostilation.pong.engine.IPongEngine
import java.io.IOException

class RequestConnectionCommand<M>(private val index: Int) : AbstractPongCommand<M>() {
    @Throws(IOException::class)
    protected override fun execute(engine: IPongEngine<M>) {
        engine.connectToGame(marker, index)
    }

}